app/src/main/java/com/android/sample/ui/mainscreen/MainScreen.kt

```diff
@@ -1,46 +1,316 @@
+              workoutId = "2",
+              name = "Summer body",
+              description = "Be ready for the summer",
+              warmup = false,
+              userIdSet = mutableSetOf("user1", "user4")),
+      )
+
   Scaffold(
-      modifier = Modifier.testTag("mainScreen"),
-      topBar = {
-        TopAppBar(
-            title = { Text("Main Screen") },
-            actions = {
-              IconButton(onClick = { navigationActions.navigateTo("Settings Screen") }) {
-                Icon(
-                    imageVector = Icons.Outlined.Settings,
-                    contentDescription = "Settings",
-                    tint = Color.Black)
-              }
-            })
-      },
+      modifier = Modifier.testTag("MainScreen"),
       content = { padding ->
-        Box(
+        Column(
             modifier = Modifier.fillMaxSize().padding(padding),
-            contentAlignment = Alignment.Center) {
-              Text("No content yet")
+            verticalArrangement = Arrangement.SpaceBetween) {
+              // Display the profile section
+              ProfileSection(
+                  account = account, profile = profile, navigationActions = navigationActions)
+
+              // Display the workout sessions
+              WorkoutSessionsSection(
+                  workouts = workouts, profile = profile, navigationActions = navigationActions)
+
+              // Display the quick workout section
+              QuickWorkoutSection(navigationActions = navigationActions)
+
+              // Display the new workout plan button
+              NewWorkoutSection(navigationActions = navigationActions)
             }
       },
-      bottomBar = {
-        BottomNavigationMenu(
-            onTabSelect = { route -> navigationActions.navigateTo(route) },
-            tabList = LIST_OF_TOP_LEVEL_DESTINATIONS,
-            selectedItem = navigationActions.currentRoute())
-      })
+      bottomBar = { BottomNavigationBar(navigationActions = navigationActions) })
+}
+
+// Composable function for the profile section with profile picture and settings
+@Composable
+fun ProfileSection(account: UserAccount, profile: Int, navigationActions: NavigationActions) {
+  Row(
+      modifier =
+          Modifier.fillMaxWidth()
+              .background(DarkBlue)
+              .padding(vertical = 16.dp)
+              .testTag("ProfileSection")) {
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            modifier = Modifier.padding(horizontal = 16.dp)) {
+              // Profile image
+              Image(
+                  painter = painterResource(id = profile),
+                  contentDescription = "Profile",
+                  modifier = Modifier.size(48.dp).clip(CircleShape).testTag("ProfilePicture"))
+              Spacer(modifier = Modifier.width(8.dp))
+              // Display user name dynamically
+              Text(
+                  text = "What's up ${account.firstName}?",
```

Comment:
Several text values are hardcoded directly in the code, which could cause problems with localization in the future.

`text = "What's up ${account.firstName}?"`

As a suggestion you could use:
`text = stringResource(id = R.string.welcome_message, account.firstName)`

Reply:
Ok I totally got you point. Thank you !